### PR TITLE
refactor: random gaussian encoder

### PR DESCRIPTION
**Pull request by theUnkownName**
Refactor: Random Gaussian Encoder

### CODE REVIEW

File: encoders/numeric/RandomGaussianEncoder/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM     """
```
Comment (hanxiao) [inline encoders/numeric/RandomGaussianEncoder/Dockerfile:1]: `FROM` is broken

File: encoders/numeric/RandomGaussianEncoder/requirements.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+    https://scikit-learn.org/stable/modules/generated/sklearn.random_projection.GaussianRandomProjection.html
```
Comment (hanxiao) [inline encoders/numeric/RandomGaussianEncoder/requirements.txt:1]: this is incorrect, it is pip requirement, not url

File: encoders/numeric/RandomGaussianEncoder/tests/test_randomgaussianencoderload.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import numpy as np
+import pickle
+
+from .. import RandomGaussianEncoder
+from .. import TransformEncoder
+
+
+class RandomGaussianEncoderLoadTestCase:
+    def test_randomgaussinencoderload(self):
```
Comment (hanxiao) [inline encoders/numeric/RandomGaussianEncoder/tests/test_randomgaussianencoderload.py:9]: better merge these two test cases together and remove `class`

File: encoders/numeric/RandomGaussianEncoder/tests/test_randomgaussianencoderload.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import numpy as np
+import pickle
+
+from .. import RandomGaussianEncoder
+from .. import TransformEncoder
+
+
+class RandomGaussianEncoderLoadTestCase:
```
Comment (JoanFM) [inline encoders/numeric/RandomGaussianEncoder/tests/test_randomgaussianencoderload.py:8]: Please adapt it to `pytest` standard.

this means.
No class, only a set of `test_....` functions.
Be aware of temporary files that need to be removed (see AnnoyIndexer tests under hub for reference)

**Review by hanxiao — CHANGES_REQUESTED** [review]
please fix the issues I highlighted

File: encoders/numeric/RandomGaussianEncoder/tests/test_randomgaussianencoderload.py (hunk 3 above)
Comment (JoanFM) [inline encoders/numeric/RandomGaussianEncoder/tests/test_randomgaussianencoderload.py:9]: be aware that these tests are not testing anything. a `test` does not have to return anything, but assert conditions. You have to check the `NumericTestCase` it was inheriting from and see what was really being tested there

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: encoders/numeric/RandomGaussianEncoder/tests/test_randomgaussianencoderload.py (hunk 3 above)
Comment (JoanFM) [reply encoders/numeric/RandomGaussianEncoder/tests/test_randomgaussianencoderload.py:9]: And please make sure something is actually tested, tests under the old test were actually being done by `NumericTestCase` base class

File: encoders/numeric/RandomGaussianEncoder/tests/test_randomgaussianencoder.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+def test_randomgaussinencodertrain():
+    requires_train_after_load = True
+    input_dim = 28
+    target_output_dim = 7
+    encoder = RandomGaussianEncoder(output_dim=target_output_dim)
+    train_data = np.random.rand(2000, input_dim)
+    encoder.train(train_data)
+
+
+def test_randomgaussinencoderload():
```
Comment (JoanFM) [inline encoders/numeric/RandomGaussianEncoder/tests/test_randomgaussianencoder.py:26]: at some point we need to test encode,

check everything that is being tested here:

```python


class NumericTestCase(JinaTestCase):
    @property
    def workspace(self):
        return os.path.join(os.environ['TEST_WORKDIR'], 'test_tmp')

    @property
    def target_output_dim(self):
        return self._target_output_dim

    @target_output_dim.setter
    def target_output_dim(self, output_dim):
        self._target_output_dim = output_dim

    @property
    def requires_train_after_load(self):
        return self._requires_train_after_load

    @requires_train_after_load.setter
    def requires_train_after_load(self, train_after_load):
        self._requires_train_after_load = train_after_load

    @property
    def input_dim(self):
        return self._input_dim

    @input_dim.setter
    def input_dim(self, input_dim):
        self._input_dim = input_dim

    @property
    def train_data(self):
        return self._train_data

    @train_data.setter
    def train_data(self, train_data):
        self._train_data = train_data

    def get_encoder(self):
        encoder = self._get_encoder()
        if encoder is not None:
            encoder.workspace = self.workspace
            self.add_tmpfile(encoder.workspace)
        return encoder

    def _get_encoder(self):
        return None

    def test_encoding_results(self):
        encoder = self.get_encoder()
        if encoder is None:
            return
        test_data = np.random.rand(10, self.input_dim)
        encoded_data = encoder.encode(test_data)
        self.assertEqual(encoded_data.shape, (test_data.shape[0], self.target_output_dim))
        self.assertIs(type(encoded_data), np.ndarray)

    def test_save_and_load(self):
        encoder = self.get_encoder()
        if encoder is None:
            return
        test_data = np.random.rand(10, self.input_dim)
        encoded_data_control = encoder.encode(test_data)
        encoder.touch()
        encoder.save()
        self.assertTrue(os.path.exists(encoder.save_abspath))
        encoder_loaded = BaseExecutor.load(encoder.save_abspath)

        if not self.requires_train_after_load:
            # some models are not deterministic when training, so even with same training data, we cannot ensure
            # same encoding results
            encoded_data_test = encoder_loaded.encode(test_data)
            np.testing.assert_array_equal(
                encoded_data_test, encoded_data_control)

    def test_save_and_load_config(self):
        encoder = self.get_encoder()
        if encoder is None:
            return
        encoder.save_config()
        self.assertTrue(os.path.exists(encoder.config_abspath))
        encoder_loaded = BaseExecutor.load_config(encoder.config_abspath)

        if self.requires_train_after_load:
            encoder_loaded.train(self.train_data)

        test_data = np.random.rand(10, self.input_dim)
        encoded_data_test = encoder_loaded.encode(test_data)
        self.assertEqual(encoded_data_test.shape, (10, self.target_output_dim))

```

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Pull request closed without merging**